edu_edfi_airflow/providers/edfi/transfers/edfi_to_s3.py

```diff
@@ -209,3 +251,109 @@ def to_jsonl_string(rows: Iterator[dict]) -> bytes:
             json.dumps(row).encode('utf8') + b'\n'
             for row in rows
         )
+
+
+class BulkEdFiToS3Operator(EdFiToS3Operator):
+    """
+    Inherits from EdFiToS3Operator to reduce code-duplication.
+
+    The following arguments MUST be lists instead of singletons:
+    - resource
+    - namespace
+    - page_size
+    - num_retries
+    - change_version_step_size
+    - query_parameters
+    - min_change_version
+    - s3_destination_filename
+
+    If all endpoints skip, raise an AirflowSkipException.
+    If at least one endpoint fails, push the XCom and raise an AirflowFailException.
+    Otherwise, return a successful XCom.
+    """
+    def execute(self, context) -> str:
+        """
+
+        :param context:
+        :return:
+        """
+        # Force destination_dir and destination_filename arguments to be used.
+        if self.s3_destination_key or not (self.s3_destination_dir and self.s3_destination_filename):
+            raise ValueError(
+                "Bulk operators require arguments `s3_destination_dir` and `s3_destination_filename` to be passed."
+            )
+
+        # Make connection outside of loop to not re-authenticate at every resource.
+        edfi_conn = EdFiHook(self.edfi_conn_id).get_conn()
+
+        # Gather DAG-level endpoints outside of loop.
+        config_endpoints = airflow_util.get_config_endpoints(context)
+
+        return_tuples = []
+        failed_endpoints = []  # Track which endpoints failed during ingestion.
+
+        # Presume all argument lists are equal length, and iterate each endpoint. Only add to the return in successful pulls.
+        zip_arguments = [
+            self.resource,
+            self.min_change_version,
+            self.namespace,
+            self.page_size,
+            self.num_retries,
+            self.change_version_step_size,
+            self.query_parameters,
+            self.s3_destination_filename,
+        ]
+
+        for idx, (resource, min_change_version, namespace, page_size, num_retries, change_version_step_size, query_parameters, s3_destination_filename) \
```

Comment:
Loop over list-arguments and complete Ed-Fi to S3 ingestions for each. Always return an xcom if any endpoints succeeded, although mark the task as failed if any failed ingestion.